lib/middleware/checkModulesAllUseSameOrigamiVersion.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+const origamiVersion1Modules = require('../data/origami-specification-version-one-module-versions.json');
+const parseModulesParameter = require('../utils/parseModulesParameter');
+const semver = require('semver');
+const httpError = require('http-errors');
+
+/**
+ * Find out the Origami Component specification version that the component version was built with.
+ *
+ * @param {string} name The name of the Origami Component whose Origami Version to return
+ * @param {string} version The version of the Origami Component whose Origami Version to return
+ * @returns {number} The Origami Version of the component named `name` at version `version`
+ */
+function getOrigamiVersion(name, version) {
+	if (origamiVersion1Modules[name]) {
+		// If the version is not a range, it is likely to be a git ref such as a branch name.
+		// If the component has a version which was origami spec v1, then we assume the
+		// git ref is going to return a version of the component which is origmai spec v1.
+		if (!semver.validRange(version)) {
+			return 1;
+		}
+
+		if (semver.intersects(origamiVersion1Modules[name], version)) {
```

Comment:
What we want is to confirm that the requested version range will _always_ return a version of the component which was built against spec v1.

`semver.intersects` does not do that, it will return `true` if the two ranges have some intersection, which means we could give it a range which can resolve to a version of the component which was built against spec v2 and it would return `true`

```
// intersects(range): Return true if any of the ranges comparators intersect
semver.intersects('<= 4', '>= 4') // returns true
```

I believe what we want is to confirm that the requested version range is a subset of the range which contains all origami spec v1 versions of the component.

```
//subset(subRange, superRange): Return true if the subRange range is entirely contained by the superRange range.
semver.subset('>=4', '<=4') // returns false
```

```suggestion
		if (semver.subset(version, origamiVersion1Modules[name])) {
```